tss-esapi/tests/context_tests/tpm_commands/duplication_commands_tests.rs

```diff
@@ -1,2 +1,193 @@
+        SymmetricDefinitionObject,
+    };
+
+    #[test]
+    fn test_duplicate() {
+        let mut context = create_ctx_without_session();
+
+        let trial_session = context
+            .start_auth_session(
+                None,
+                None,
+                None,
+                SessionType::Trial,
+                SymmetricDefinition::AES_256_CFB,
+                HashingAlgorithm::Sha256,
+            )
+            .expect("Start auth session failed")
+            .expect("Start auth session returned a NONE handle");
+
+        let (policy_auth_session_attributes, policy_auth_session_attributes_mask) =
+            SessionAttributesBuilder::new()
+                .with_decrypt(true)
+                .with_encrypt(true)
+                .build();
+        context
+            .tr_sess_set_attributes(
+                trial_session,
+                policy_auth_session_attributes,
+                policy_auth_session_attributes_mask,
+            )
+            .expect("tr_sess_set_attributes call failed");
+
+        let policy_session = PolicySession::try_from(trial_session)
+            .expect("Failed to convert auth session into policy session");
+
+        context
+            .policy_auth_value(policy_session)
+            .expect("Policy auth value");
+
+        context
+            .policy_command_code(policy_session, TPM2_CC_Duplicate)
+            .expect("Policy command code");
+
+        let digest = context
+            .policy_get_digest(policy_session)
+            .expect("Could retrieve digest");
+
+        drop(context);
+        let mut context = create_ctx_with_session();
+
+        let parent_of_object_to_duplicate_handle = context
+            .create_primary(
+                Hierarchy::Owner,
+                &decryption_key_pub(),
+                None,
+                None,
+                None,
+                None,
+            )
+            .unwrap()
+            .key_handle;
+
+        // Fixed TPM and Fixed Parent should be "false" for an object
+        // to be elligible for duplication
+        let object_attributes = ObjectAttributesBuilder::new()
+            .with_fixed_tpm(false)
+            .with_fixed_parent(false)
+            .with_sensitive_data_origin(true)
+            .with_user_with_auth(true)
+            .with_decrypt(true)
+            .with_sign_encrypt(true)
+            .with_restricted(false)
+            .build()
+            .expect("Attributes to be valid");
+
+        let public_child = PublicBuilder::new()
+            .with_public_algorithm(PublicAlgorithm::Rsa)
+            .with_name_hashing_algorithm(HashingAlgorithm::Sha256)
+            .with_object_attributes(object_attributes)
+            .with_auth_policy(&digest)
+            .with_rsa_parameters(
+                PublicRsaParametersBuilder::new()
+                    .with_scheme(RsaScheme::Null)
+                    .with_key_bits(RsaKeyBits::Rsa2048)
```

Comment:
If you want your test to be different from your documentation then one way of doing it is to use another type of key like a ECC key.

Reply:
Excellent idea! I'll rework the test case to use ECC keys.